api/geyser/src/main/java/org/geysermc/geyser/api/custom/items/FullyCustomItemData.java

```diff
@@ -27,316 +27,173 @@
-    }
-
-    /**
-     * Sets the repair materials of the item.
-     *
-     * @param repairMaterials the repair materials of the item
-     */
-    public void repairMaterials(@Nullable Set<String> repairMaterials) {
-        this.repairMaterials = repairMaterials;
-    }
+    @Nullable Set<String> repairMaterials();
 
     /**
      * Gets if the item has the suspicious stew effect.
      *
      * @return if the item has the suspicious stew effect
      */
-    public boolean hasSuspiciousStewEffect() {
-        return this.hasSuspiciousStewEffect;
-    }
-
-    /**
-     * Sets if the item has the suspicious stew effect.
-     *
-     * @param suspiciousStewEffect if the item has the suspicious stew effect
-     */
-    public void hasSuspiciousStewEffect(boolean suspiciousStewEffect) {
-        this.hasSuspiciousStewEffect = suspiciousStewEffect;
-    }
+    boolean hasSuspiciousStewEffect();
 
     /**
      * Gets the item's creative category, or tab id.
      *
      * @return the item's creative category
      */
-    public @Nullable Integer creativeCategory() {
-        return this.creativeCategory;
-    }
-
-    /**
-     * Sets the item's creative category, or tab id.
-     *
-     * @param creativeCategory the item's creative category
-     */
-    public void creativeCategory(@Nullable Integer creativeCategory) {
-        this.creativeCategory = creativeCategory;
-    }
+    @Nullable Integer creativeCategory();
 
     /**
      * Gets the item's creative group.
      *
      * @return the item's creative group
      */
-    public @Nullable String creativeGroup() {
-        return this.creativeGroup;
-    }
-
-    /**
-     * Sets the item's creative group.
-     *
-     * @param creativeGroup the item's creative group
-     */
-    public void creativeGroup(@Nullable String creativeGroup) {
-        this.creativeGroup = creativeGroup;
-    }
+    @Nullable String creativeGroup();
 
     /**
      * Gets if the item is a hat. This is used to determine if the item should be rendered on the player's head, and
      * normally allow the player to equip it. This is not meant for armor.
      *
      * @return if the item is a hat
      */
-    public boolean isHat() {
-        return this.isHat;
-    }
-
-    /**
-     * Sets if the item is a hat. This is used to determine if the item should be rendered on the player's head, and
-     * normally allow the player to equip it. This is not meant for armor.
-     *
-     * @param isHat if the item is a hat
-     */
-    public void isHat(boolean isHat) {
-        this.isHat = isHat;
-    }
+    boolean isHat();
 
     /**
      * Gets if the item is a tool. This is used to set the render type of the item, if the item is handheld.
      *
      * @return if the item is a tool
      */
-    public boolean isTool() {
-        return this.isTool;
+    boolean isTool();
+
+    static FullyCustomItemData.Builder builder(String name, String identifier, int javaId) {
```

Comment:
Is there any reason these are provided as method parameters rather than being part of the builder?

Reply:
Those are required parameters for the item registration.